sqs_lambda/index.ts

```diff
@@ -132,87 +61,58 @@ export async function handlerFn(event: SQSEvent) {
         );
       }
 
-      if (fxaEvent.event === EVENT.USER_DELETE) {
-        const res = await submitDeleteMutation(fxaEvent.user_id);
-        if (res?.errors) {
-          if (res.errors.CODE && res.errors.CODE == 'NOT_FOUND') {
-            console.info('FxA User not found', {
-              userId: fxaEvent.user_id,
-              event: fxaEvent,
-            });
-          } else {
+      switch (fxaEvent.event) {
+        case EVENT.APPLE_MIGRATION:
+          if (!fxaEvent.user_email) {
             throw new Error(
-              `Error processing ${record.body}: \n${JSON.stringify(
-                res?.errors
-              )}`
+              `Error processing ${record.body}: missing user_email`
             );
           }
-        }
-      }
-
-      if (fxaEvent.event === EVENT.APPLE_MIGRATION) {
-        if (!fxaEvent.user_email) {
-          throw new Error(
-            `Error processing ${record.body}: missing user_email`
-          );
-        }
 
-        if (!fxaEvent.transfer_sub) {
-          throw new Error(
-            `Error processing ${record.body}: missing transfer_sub`
-          );
-        }
-
-        const res = await migrateAppleUserMutation(
-          fxaEvent.user_id,
-          fxaEvent.user_email,
-          fxaEvent.transfer_sub
-        );
-        if (res?.errors) {
-          if (res.errors.CODE && res.errors.CODE == 'NOT_FOUND') {
-            console.info('FxA User not found', {
-              userId: fxaEvent.user_id,
-              event: fxaEvent,
-            });
-          } else {
+          if (!fxaEvent.transfer_sub) {
             throw new Error(
-              `Error processing ${record.body}: \n${JSON.stringify(
-                res?.errors
-              )}`
+              `Error processing ${record.body}: missing transfer_sub`
             );
           }
-        }
-      }
 
-      if (fxaEvent.event === EVENT.PROFILE_UPDATE) {
-        // only handling email updates in this block for this event
-        // early exit if no email property present
-        if (!isEmailUpdatedEvent(fxaEvent)) {
-          return;
-        }
+          res = await migrateAppleUserMutation(
+            fxaEvent.user_id,
+            fxaEvent.user_email,
+            fxaEvent.transfer_sub
+          );
 
-        const res = await submitEmailUpdatedMutation(
-          fxaEvent.user_id,
-          fxaEvent.user_email
-        );
+          break;
+        case EVENT.PASSWORD_CHANGE:
+          res = await passwordChangeMutation(fxaEvent.user_id);
 
-        if (res?.errors) {
-          if (res.errors.CODE && res.errors.CODE == 'NOT_FOUND') {
-            console.info('FxA User not found', {
-              userId: fxaEvent.user_id,
-              event: fxaEvent,
-            });
-          } else {
-            throw new Error(
-              `Error processing ${record.body}: \n${JSON.stringify(
-                res?.errors
-              )}`
-            );
```

Comment:
these repeated `if (res?.errors)` blocks were abstracted into a single function in `mutations.ts`.

Reply:
YAY